### PR TITLE
Modify S6335: Remove FPs

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml [hunk 1]
```diff
@@ -2,43 +2,33 @@ provider:
...
     - rspecKey: S6335
       id: google-service-account-keys
       metadata:
         name: Google Service Account Keys
+      detection:
+        # Matches multiline private keys or single line private keys containing '\n's
+        matching:
+          pattern: "(\
+                    -----BEGIN PRIVATE KEY-----\
+                    (?:\\n|\\\\{1,2}n)\
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml:27]: Maybe match also other whitespaces here? The key could be indented. You could also first match for the line break and only afterwards accept other whitespaces.
```suggestion
                    (?:\\s|\\\\{1,2}n)++\
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml [hunk 2]
```diff
@@ -2,43 +2,33 @@ provider:
...
+                    (?:[0-9A-Za-z/+]{80,}|(?:[0-9A-Za-z/+]{40,}(?:\\n|\\\\{1,2}n))+[0-9A-Za-z/+=]+)={0,2}\
+                    (?:\\n|\\\\{1,2}n)\
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml:29]: Same as above about potential indentation.

File: sonar-text-plugin/src/test/java/org/sonar/plugins/common/TextAndSecretsSensorTest.java [hunk 3]
```diff
@@ -73,20 +73,6 @@ void describe() {
     assertThat(logTester.logs()).isEmpty();
   }
 
-  @Test
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/test/java/org/sonar/plugins/common/TextAndSecretsSensorTest.java:76]: Why is this removed? Was it removed automatically when running the script to create the tests?

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
Overall good. Minor request about indentation.
Also if you want you can think about adjusting the regex to also find stuff like:
```
private_key = "-----BEGIN PRIVATE KEY----- " \
            + "asdjasdjkasdnasd..." \
            + "-----END PRIVATE KEY-----";
```
Of course, this might look differently in different languages so I am not sure if this would be too much effort and not the goal of this PR right now.

File: sonar-text-plugin/src/test/java/org/sonar/plugins/common/TextAndSecretsSensorTest.java (hunk 3 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/test/java/org/sonar/plugins/common/TextAndSecretsSensorTest.java:76]: No, I removed it because it is an artifact from the beginnings of sonar-text

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml (hunk 1 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gcp.yaml]: done !

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM!
I would suggest to rename the test file to GCPCheckTest.java since this is probably the name our tool would give it automatically. I think it is usually {name_of_the_provider}CheckTest.java
